File: backend/src/main/java/com/eventeasy/service/DashboardService.java
package com.eventeasy.service;

import com.eventeasy.dto.response.DashboardStatsResponse;

/**
 * <p>Service interface declaring executive dashboard analytics operations.</p>
 *
 * @author EventEasy Engineering
 * @version 1.0.0
 */
public interface DashboardService {

    /**
     * Generate complete Dashboard metrics, Recharts distribution charts, and recent enquiry feed.
     *
     * @return DashboardStatsResponse DTO
     */
    DashboardStatsResponse getDashboardStats();
}

File: frontend/src/services/dashboardService.js
import axiosClient from '../api/axiosClient';

/**
 * Dashboard Analytics API Service wrapper.
 */
export const dashboardService = {
  getStats: async () => {
    const res = await axiosClient.get('/dashboard/stats');
    return res.data; // DashboardStatsResponse
  },
};
